ldap/servers/slapd/eventq.c

```diff
@@ -423,12 +451,22 @@ eq_stop()
      * it acknowledges by setting eq_stopped.
      */
     while (!eq_stopped) {
-        PR_Lock(eq->eq_lock);
-        PR_NotifyAllCondVar(eq->eq_cv);
-        PR_Unlock(eq->eq_lock);
-        PR_Lock(ss_lock);
-        PR_WaitCondVar(ss_cv, PR_MillisecondsToInterval(100));
-        PR_Unlock(ss_lock);
+        struct timespec current_time = {0};
+
+        pthread_mutex_lock(&(eq->eq_lock));
+        pthread_cond_broadcast(&(eq->eq_cv));
+        pthread_mutex_unlock(&(eq->eq_lock));
+
+        pthread_mutex_lock(&ss_lock);
+        clock_gettime(CLOCK_MONOTONIC, &current_time);
+        if (current_time.tv_nsec + 100000000 > 1000000000) {
+            /* ns will overflow, just bump the seconds */
+            current_time.tv_sec++;
```

Comment:
I don't think we can do that because we need to move the clock forward, this could move it backwards because we are not checking the current value of nsecs.  nsecs could already be at 999999999, so changing nano seconds to 1 second minus the delay would move the clock backwards and that would negate the timed wait.  

Now the current code is not perfect, it's not always going to sleep 100 ms if there is an overflow of nanoseconds.  I will revise this...

Reply:
Yes the change looks good. I had a doubt regarding  the lost of nsec in case of nsec overflow.